### PR TITLE
Add example of parsing yaml into unstructured types

### PR DESCRIPTION
**Pull request by justinsb**
Adding an example to demonstrate the/a recommended way to parse
objects.

/kind documentation

```release-note
NONE
```

### CODE REVIEW

**Comment by justinsb** [comment]
I needed pointers from @liggitt on how to do this, and this is my attempt at writing up a strawman.

@deads2k apparently you did a lot of the work to make this possible in kubectl, so @liggitt suggested you might be a good person to tell me where I went astray!

File: staging/src/k8s.io/client-go/examples/unstructured-parsing/main_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+spec:
+    bar: 2
+
+metadata:
+  name: bar
+
+`
+
+// The expected output - note the normalization of spacing & field order
+var expected = `---
```
Comment (justinsb) [inline staging/src/k8s.io/client-go/examples/unstructured-parsing/main_test.go]: The `---` is a little annoying here - is there a trick here?

File: staging/src/k8s.io/client-go/examples/unstructured-parsing/main_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+metadata:
+  name: bar
+spec:
+  bar: 2
+`
+
+// TestUnstructuedParsing demonstrates how to parse a yaml file, using the unstructured codecs.
+func TestUnstructuredParsing(t *testing.T) {
+	// Build a yaml decoder with the unstructured Scheme
+	yamlDecoder := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
```
Comment (justinsb) [inline staging/src/k8s.io/client-go/examples/unstructured-parsing/main_test.go]: I wasn't sure if there was a way to auto-detect json vs yaml and whether I actually should be using a different API entirely

**Comment by discordianfish** [comment]
@justinsb I'm trying to do something similar by using `unstructured.UnstructuredJSONScheme.Decode(content, nil, nil)`, still need to figure out how to handle yaml but seems like the better way?

There is also `scheme.Codecs.UniversalDeserializer().Decode` which handles yaml automagically but that requires me to register available schemas at compiel time.

**Comment by discordianfish** [comment]
btw, I've figured it out: https://stackoverflow.com/questions/53341727/how-to-submit-generic-runtime-object-to-kubernetes-api-using-client-go/55287249#55287249

**Pull request closed without merging**